### PR TITLE
fix(explore): DndColumnSelect not handling controls with "multi: false"

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
`DndColumnSelect` was working correctly only for controls that accept multiple values. When used with controls that had `multi: false` prop, it displayed a prompt to add more columns and behaved weirdly - sometimes it allowed adding more values, sometimes it replaced the first value with whatever user dropped last. This PR fixes those glitches and unifies the behaviour with with `DndMetricSelect` - after the first value is added, prompt disappears and adding new columns is disabled.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:

https://user-images.githubusercontent.com/67837651/119038421-f352ec00-b967-11eb-96e4-16daf1f439ee.mov

After:

https://user-images.githubusercontent.com/67837651/119038451-fa79fa00-b967-11eb-952a-c110f298d8cf.mov

### TEST PLAN
Open a chart that uses `DndColumnSelect` for single value controls (e.g. Country Map and "ISO 3166-2 CODES" control), verify that only 1 value can be added and there are no glitches when you try to add more values. Please don't mind type label overlapping with column name, the fix is ready in a different PR 🙂 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

CC: @villebro @junlincc

### CODE REVIEW

**Review by michael-s-molina — APPROVED** [review]

**Review by villebro — APPROVED** [review]
LGTM - one small improvement proposal

File: superset-frontend/src/explore/components/controls/DndColumnSelectControl/DndColumnSelect.tsx [hunk 1]
```diff
@@ -77,6 +78,7 @@ export const DndColumnSelect = (props: LabelProps) => {
       canDrop={canDrop}
       valuesRenderer={valuesRenderer}
       accept={DndItemType.Column}
+      displayGhostButton={multi || optionSelector.values.length === 0}
```
[the remark this answers is not in the document]
Comment (kgabryje) [reply superset-frontend/src/explore/components/controls/DndColumnSelectControl/DndColumnSelect.tsx:82]: Thanks! I also changed it in `DndMetricSelect`
image

**Comment by junlincc** [comment]
@michael-s-molina @villebro thanks for reviewing! did you test locally? for some reasons i still see the select fields when i pull the PR

**Comment by kgabryje** [comment]
> @michael-s-molina @villebro thanks for reviewing! did you test locally? for some reasons i still see the select fields when i pull the PR

@junlincc Try on Country Map chart. I recorded those videos while implementing dnd controls for all charts (not available on master yet, PR coming in ~30 minutes 🙂)

**Pull request merged**